addon/services/cookies.js

```diff
@@ -73,35 +77,21 @@ export default Ember.Service.extend({
   },
 
   _writeDocumentCookie(name, value, options = {}) {
-    let cookie = `${name}=${value}`;
-
-    if (!isEmpty(options.domain)) {
-      cookie = `${cookie}; domain=${options.domain}`;
-    }
-    if (typeOf(options.expires) === 'date') {
-      cookie = `${cookie}; expires=${options.expires.toUTCString()}`;
-    }
-    if (!isEmpty(options.maxAge)) {
-      cookie = `${cookie}; max-age=${options.maxAge}`;
-    }
-    if (!!options.secure) {
-      cookie = `${cookie}; secure`;
-    }
-    if (!isEmpty(options.path)) {
-      cookie = `${cookie}; path=${options.path}`;
-    }
-    this.set('_document.cookie', cookie);
+    let serializedCookie = this._serializeCookie(name, value, options);
+    this.set('_document.cookie', serializedCookie);
   },
 
   _writeFastBootCookie(name, value, options = {}) {
+    let responseHeaders  = this.get('_fastBoot.response.headers');
+    let serializedCookie = this._serializeCookie(...arguments);
+
     if (!isEmpty(options.maxAge)) {
       options.maxAge = options.maxAge * 1000;
     }
 
     this._cacheFastBootCookie(...arguments);
 
-    let response = this.get('_fastBoot._fastBootInfo.response');
-    response.cookie(name, value, options);
+    responseHeaders.append('cookie', serializedCookie);
```

Comment:
👍  will fix. [http cookies reference](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies) for posterity.